website/common/script/ops/unlock.js

```diff
@@ -36,27 +73,17 @@ export default function unlock (user, req = {}, analytics) {
     cost = 0.5;
   }
 
-  let setPaths;
+  const setPaths = path.split(',');
   let alreadyOwns;
 
   if (isFullSet) {
-    setPaths = path.split(',');
-    let alreadyOwnedItems = 0;
-
-    each(setPaths, singlePath => {
-      if (get(user, `purchased.${singlePath}`) === true) {
-        alreadyOwnedItems += 1;
-      }
-    });
-
-    if (alreadyOwnedItems === setPaths.length) {
+    const alreadyOwnedItems = setPaths.filter(p => get(user, `purchased.${p}`)).length;
+    const totalItems = setPaths.length;
+    if (alreadyOwnedItems === totalItems) {
       throw new NotAuthorized(i18n.t('alreadyUnlocked', req.language));
-    // TODO write math formula to check if buying
-    // the full set is cheaper than the items individually
-    // (item cost * number of remaining items) < setCost`
-    } /* else if (alreadyOwnedItems > 0) {
+    } else if ((totalItems - alreadyOwnedItems) < 3) {
```

Comment:
For this one I'd prefer to handle it in the next PR so that we can have a precise check on the number of items in the set and their cost since we want to be flexible about how many items there are in a set. In the PR I'll make right after this one I'll be sure to check all the sets against the content api for their price and the items in them

Reply:
But the price for all sets is actually set in the code to be 15 gems for background sets and 5 gems for any other set. That combined with the fact that backgrounds cost 7 gems per single and other items cost 2 gems per single means that currently, for all uses of the unlock functionality you need to buy less than 3 new items at once in order for single purchases to be cheaper.

However, I'll re-comment it out and make a follow-up pull request. No need to burden this PR even more 😄.